### PR TITLE
Fix a few issues around readTexture and iOS camera

### PR DESCRIPTION
**Pull request by Alex-MSFT**
This PR fixes a few issues related to the native camera implementation on iOS and readPixels
1. The current implementation for readTextureAsync assumes that the frame number will always increment by 1 when DeviceImpl::Frame is called.  This would be true if we only ever call bgfx::frame in DeviceImpl::Frame, but we have at least two other call sites for bgfx::frame, so this is not a true assumption.  In general this implementation was prone to the queue getting blocked, so I'm making it a bit more lenient.  (See #1131 for a related issue).
2. NativeCameraImpl was not creating a new ImplData object when calling reset() which meant that the unique pointer was invalid on subesquent NativeCamera open calls, and leading to an AV.
3. On Apple devices when initializing a video texture in landscape mode on we were never creating the textureRGBA object, because the resolution from the camera already matched the initial camera dimensions, so I am explicitly setting the textureRGBA value to null when calling Open and validating that we have a valid texture at the start of updateCameraTexture.
4. iOS was using dispatch_sync when creating video textures, this can cause deadlocks on iOS if already running on the main thread or if the main thread is blocked on the current thread.  Changed this to dispatch_async instead.

### CODE REVIEW

File: Core/Graphics/Source/DeviceImpl.cpp [hunk 1]
```diff
@@ -390,8 +398,7 @@ namespace Babylon::Graphics
         uint32_t frameNumber{bgfx::frame()};
 
         // Process read texture requests.
-        assert(m_readTextureRequests.empty() || m_readTextureRequests.front().first >= frameNumber);
-        while (!m_readTextureRequests.empty() && m_readTextureRequests.front().first == frameNumber)
+        while (!m_readTextureRequests.empty() && m_readTextureRequests.front().first <= frameNumber)
```
Comment (CoPrez) [inline Core/Graphics/Source/DeviceImpl.cpp:401]: Are the readPixels pending promises in this situation still valid? Or should they just be error'd out like you do above?

**Review by CoPrez — APPROVED** [review]

File: Core/Graphics/Source/DeviceImpl.cpp (hunk 1 above)
Comment (Alex-MSFT) [reply Core/Graphics/Source/DeviceImpl.cpp:401]: As far as I can tell the results are still valid.  I'll set up a test to explicitly delay the sending of results to double check.

Comment (Alex-MSFT) [reply Core/Graphics/Source/DeviceImpl.cpp:401]: The results are still valid up to at least 1000 frames later.  I don't think we'll ever hit a scenario where this goes longer than 1-2 frames over.

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm [hunk 2]
```diff
@@ -457,7 +463,7 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
 
     void Camera::Impl::Close()
     {
-        m_implData.reset();
+        m_implData.reset(new ImplData);
```
Comment (ryantrem) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:466]: Why does this have to be destroyed? And if it does have to be destroyed, can we instead do `m_implData = std::make_unique<ImplData>()`?

**Pull request merged**